test/lib/mayaUsd/render/mayaToHydra/testStageVariants.py

```diff
@@ -0,0 +1,119 @@
+        cubesPath = ufe.Path([
+            mayaUtils.createUfePathSegment("|variants|variantsShape"), 
+            usdUtils.createUfePathSegment("/Cubes")])
+        cubesItems = ufe.Hierarchy.createItem(cubesPath)
+        self.assertIsNotNone(cubesItems)
+        ufe.GlobalSelection.get().clear()
+        ufe.GlobalSelection.get().append(cubesItems)
+        self.assertSnapshotClose("oneCube.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT)
+        ufe.GlobalSelection.get().clear()
+        
+        modVariant.SetVariantSelection('TwoCubes')
+        self.assertEqual(modVariant.GetVariantSelection(), 'TwoCubes')
+        ufe.GlobalSelection.get().append(cubesItems)
+        self.assertSnapshotClose("twoCubes.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT)
+        ufe.GlobalSelection.get().clear()
+        
+        modVariant.SetVariantSelection('ThreeCubes')
+        self.assertEqual(modVariant.GetVariantSelection(), 'ThreeCubes')
+        ufe.GlobalSelection.get().append(cubesItems)
+        self.assertSnapshotClose("threeCubes.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT)
+        ufe.GlobalSelection.get().clear()
+        
+        #Check displacement variant for CubeOne (not an actual displacement but a different transform)
+        cube1Prim = shapeStage.GetPrimAtPath('/Cubes/Geom/CubeOne')
+        self.assertFalse(cube1Prim== None)
+        displacementVariant = cube1Prim.GetVariantSet('displacement')
+        self.assertFalse(displacementVariant== None)
+        self.assertEqual(displacementVariant.GetVariantSelection(), 'none')
+        displacementVariant .SetVariantSelection('moved')
+        self.assertEqual(displacementVariant.GetVariantSelection(), 'moved')
+        ufe.GlobalSelection.get().append(cubesItems)
+        self.assertSnapshotClose("threeCubesWithDisplacement.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT)
+        ufe.GlobalSelection.get().clear()
+
+        #Check if it works when the prims are instanceable
+        cube2Prim = shapeStage.GetPrimAtPath('/Cubes/Geom/CubeTwo')
+        self.assertFalse(cube2Prim== None)
+        cube3Prim = shapeStage.GetPrimAtPath('/Cubes/Geom/CubeThree')
+        self.assertFalse(cube3Prim== None)
+        cube1Prim.SetInstanceable(True)
+        self.assertTrue(cube1Prim.IsInstanceable())
+        cube2Prim.SetInstanceable(True)
+        self.assertTrue(cube2Prim.IsInstanceable())
+        cube3Prim.SetInstanceable(True)
+        self.assertTrue(cube3Prim.IsInstanceable())
+        modVariant.SetVariantSelection('OneCube')
+        self.assertEqual(modVariant.GetVariantSelection(), 'OneCube')
+        ufe.GlobalSelection.get().append(cubesItems)
+        self.assertSnapshotClose("oneCubeInstanceable.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT)
+        ufe.GlobalSelection.get().clear()
+
+        #Enable the following tests when HYDRA-820 is fixed
```

Comment:
Hope you have the task of enabling this test as an acceptance criterion in HYDRA-820 :)

Reply:
Yes, I've added a TODO in the task.